stan/math/gpu/kernels/lower_tri_inverse_step2.hpp

```diff
@@ -0,0 +1,122 @@
+#ifndef STAN_MATH_GPU_KERNELS_LOWER_TRI_INVERSE_HPP
+#define STAN_MATH_GPU_KERNELS_LOWER_TRI_INVERSE_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/gpu/kernel_cl.hpp>
+
+namespace stan {
+namespace math {
+namespace opencl_kernels {
+// \cond
+const char* lower_tri_inverse_step2_kernel_code = STRINGIFY(
+    // \endcond
+    /**
+     * Calculates the intermediate products in the lower_tri_inverse
+     *
+     * @param[in] A input matrix that is being inverted.
+     * @param[out] C temporary matrix to store the intermediate results.
+     * @param M The number of rows for A.
+     * @param temp_rows The number of elements to multiply
+     * @param non_padded_rows The number of rows that are not used for padding.
+     * @note Code is a <code>const char*</code> held in
+     * <code>lower_tri_inverse_step2_kernel_code.</code>
+     *  Used in math/gpu/lower_tri_inverse.hpp.
+     *  This kernel uses the helper macros available in helpers.cl.
+     */
+    __kernel void lower_tri_inverse_step2(
+        __global read_only double* A, __global double* C, const int M,
+        const int temp_rows, int non_padded_rows) {
+      int t = get_global_id(2);
+      int offset = t * temp_rows * 2;
+      // thread index inside the thread_block
+      const int thread_block_row = get_local_id(0);
+      const int thread_block_col = get_local_id(1);
+      // global thread index
+      const int i = THREAD_BLOCK_SIZE * get_group_id(0) + thread_block_row;
```

Comment:
I think of `i` and `j` being reserved for iterator variables that keep track of loop indices. Can we give these descriptive names since they're longer-lived and not used that way?